packages/block-library/src/template-part/edit/index.js

```diff
@@ -83,7 +83,7 @@ function TemplatesList( { area, clientId, isEntityAvailable, onSelect } ) {
 	const canReplace =
 		isEntityAvailable &&
 		!! blockPatterns.length &&
-		( area === 'header' || area === 'footer' );
+		( area === 'header' || area === 'footer' || area === 'overlay' );
```

Comment:
Nit: might be more concise to use an `.includes()` at this point:

```suggestion
		( [ 'header', 'footer', 'overlay' ].includes( area ) );
```

Reply:
We might want to do a check that the experiment is on before including it. Technically, the overlay template part editing shouldn't be available if the experiment is on, so maybe it's overly-defensive to leave the experiment check off.

Reply:
good call, @jeryj